Refresh.GameServer/Services/LevelListOverrideService.cs

```diff
@@ -37,14 +38,61 @@ public void AddOverridesForUser(GameUser user, IEnumerable<GameLevel> levels)
         this._userIdsToLevelList.Add(user.UserId, ids);
     }
 
-    public IEnumerable<GameLevel> GetOverridesForUser(GameUser user, GameDatabaseContext database)
+    public IEnumerable<GameLevel> GetOverridesForUser(Token token, GameDatabaseContext database)
     {
+        GameUser user = token.User;
+        
         Debug.Assert(this.UserHasOverrides(user), "User does not have overrides, should be checked first");
         
         List<int> overrides = this._userIdsToLevelList[user.UserId];
         this.Logger.LogDebug(RefreshContext.LevelListOverride, "Getting level override for {0}: [{1}]", user.Username, string.Join(", ", overrides));
         this._userIdsToLevelList.Remove(user.UserId);
 
-        return overrides.Select(database.GetLevelById)!;
+        List<GameLevel> levels = [];
+        foreach (GameLevel level in overrides.Select(levelId => database.GetLevelById(levelId)!))
+        {
+            switch (token.TokenGame)
+            {
+                //LBP1 can only play LBP1 levels.
+                case TokenGame.LittleBigPlanet1:
+                    if (level.GameVersion != TokenGame.LittleBigPlanet1) 
+                        continue;
+                            
+                    break;
+                //LBP2 can play LBP1,2
+                case TokenGame.LittleBigPlanet2:
+                    if (level.GameVersion != TokenGame.LittleBigPlanet1 && level.GameVersion != TokenGame.LittleBigPlanet2) 
+                        continue;
+                            
+                    break;
+                //LBP3 can play LBP1,2,3
+                case TokenGame.LittleBigPlanet3:
+                    if (level.GameVersion != TokenGame.LittleBigPlanet1 && level.GameVersion != TokenGame.LittleBigPlanet2 && level.GameVersion != TokenGame.LittleBigPlanet3)
+                        continue;
+                            
+                    break;
+                //LBPV can only play LBPV levels.
+                case TokenGame.LittleBigPlanetVita:
+                    if (level.GameVersion != TokenGame.LittleBigPlanetVita)
+                        continue;
+                            
+                    break;
+                //LBP PSP can only play LBP PSP levels.
+                case TokenGame.LittleBigPlanetPSP:
+                    if (level.GameVersion != TokenGame.LittleBigPlanetPSP)
+                        continue;
+                            
+                    break;
+                //Allow all for website
+                case TokenGame.Website:
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+            
+            levels.Add(level);
+        }
```

Comment:
This definitely shouldn't belong here. Pretty sure we have this logic duplicated multiple times across the codebase - maybe we can move it somewhere static? Something like `bool IsVersionCompatible(GameVersion currentGame, GameVersion levelGame)`

Reply:
I swore this was the case aswell, but [this](https://github.com/LittleBigRefresh/Refresh/blob/e52849d7841c23ee60a8eab8b07feccd48e73962/Refresh.GameServer/Extensions/LevelEnumerableExtensions.cs#L11) is the closest duplicate code i can find (but even then a static function wouldnt help for LevelEnumerableExtensions, since IQueryable shenanigans)

But extracting this code into a static function i will do anyway for the future incase we need this logic again (quite likely)